pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -0,0 +1,511 @@
+        kernel_name (str): name of transition kernel. The current version supports
+            two kernels: ``"Local"`` and ``"NonZeroRandom"``.
+            The local kernel conducts a bit-flip local transition between states.
+            The local kernel generates a random qubit site and then generates a random
+            number to determine the new bit at that qubit site. The ``"NonZeroRandom"`` kernel
+            randomly transits between states that have nonzero probability.
+        num_burnin (int): number of steps that will be dropped. Increasing this value will
+            result in a closer approximation but increased runtime.
+        batch_obs (bool): Determine whether we process observables in parallel when
+            computing the jacobian. This value is only relevant when the lightning
+            qubit is built with OpenMP.
+    """
+
+    _device_options = ("rng", "c_dtype", "batch_obs", "mcmc", "kernel_name", "num_burnin")
+    _CPP_BINARY_AVAILABLE = LQ_CPP_BINARY_AVAILABLE
+    _new_API = True
+
+    operations = _operations
+    """The names of the supported operations."""
+
+    observables = _observables
+    """The names of the supported observables."""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        wires,
+        *,
+        c_dtype=np.complex128,
+        shots=None,
+        seed="global",
+        mcmc=False,
+        kernel_name="Local",
+        num_burnin=100,
+        batch_obs=False,
+    ):
+        if not LQ_CPP_BINARY_AVAILABLE:
+            raise ImportError(
+                "Pre-compiled binaries for lightning.qubit are not available. "
+                "To manually compile from source, follow the instructions at "
+                "https://pennylane-lightning.readthedocs.io/en/latest/installation.html."
+            )
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._statevector = LightningStateVector(num_wires=len(self.wires), dtype=c_dtype)
+        seed = np.random.randint(0, high=10000000) if seed == "global" else seed
+        self._rng = np.random.default_rng(seed)
+
+        self._c_dtype = c_dtype
+        self._batch_obs = batch_obs
+        self._mcmc = mcmc
+        if self._mcmc:
+            if kernel_name not in [
+                "Local",
+                "NonZeroRandom",
+            ]:
+                raise NotImplementedError(
+                    f"The {kernel_name} is not supported and currently "
+                    "only 'Local' and 'NonZeroRandom' kernels are supported."
+                )
+            if num_burnin >= shots:
+                raise ValueError("Shots should be greater than num_burnin.")
+            self._kernel_name = kernel_name
+            self._num_burnin = num_burnin
+        else:
+            self._kernel_name = None
+            self._num_burnin = None
+
+    @property
+    def c_dtype(self):
+        """State vector complex data type."""
+        return self._c_dtype
```

Comment:
We might add `r_dtype` too, to simplify tests.

Reply:
I added a `LightningBase.dtype` and `LightningQubit2.dtype` property which is now being used in tests.

Reply:
Cool! Thanks!